crates/holochain/src/core/sys_validate/error.rs

```diff
@@ -210,6 +210,8 @@ pub enum PrevActionErrorKind {
     InvalidRoot,
     #[error("Root of source chain must have a timestamp greater than the Dna's origin_time")]
     InvalidRootOriginTime,
+    #[error("No more actions are allowed after a chain close")]
+    ActionAfterChainClose,
```

Comment:
Should there be a parallel validation that says a ChainOpen should not be allowed after any non-genesis action?

Reply:
I'm not sure we should restrict this until we work out how it's going to be used. If it's used as part of init and done first then we could check that. But if we decide we want to support migrations being done after init, through some other zome function call, then this wouldn't necessarily appear right after genesis